modules/formulize/include/formdisplay.php

```diff
@@ -395,8 +395,13 @@ function displayForm($formframe, $entry="", $mainform="", $done_dest="", $button
 
 	$mid = getFormulizeModId();
 
-	$currentURL = getCurrentURL();
-
+	$currentURL = ( (isset($_GET['ve']) && isset($_GET['fid'])) ? $_SERVER['PHP_SELF'] . "?fid=" . $_GET['fid'] : getCurrentURL() );
+	if (isset($_GET['sid'])) {
+	    list($fid) = icms::$xoopsDB->fetchRow(icms::$xoopsDB->query("SELECT fid FROM " . icms::$xoopsDB->prefix('formulize_screen') . " WHERE sid=" . $_GET['sid']));

```

Comment:
There should not be an SQL query written in this file. All SQL queries that deal with forms and screens should be in the form or screen object class. A better way to do this is to load the screen object with the right sid value, then access the fid property.